src/hotspot/share/c1/c1_Canonicalizer.cpp

```diff
@@ -469,9 +469,6 @@ void Canonicalizer::do_CompareOp      (CompareOp*       x) {
 
 
 void Canonicalizer::do_IfOp(IfOp* x) {
-  // Caution: do not use do_Op2(x) here for now since
-  //          we map the condition to the op for now!
-  move_const_to_right(x);
 }
```

Comment:
If you are saying this method can never be called, then shouldn't we make it report an error instead of being empty?

Reply:
Make sense.